src/NHibernate/Util/SerializableSystemType.cs

```diff
@@ -5,18 +5,17 @@
 namespace NHibernate.Util
 {
 	[Serializable]
-	internal sealed class SerializableSystemType : ISerializable
+	internal class SerializableSystemType : ISerializable, IEquatable<SerializableSystemType>
```

Comment:
At first I had tried simplifying it by using `OnSerializing` and `OnDeserialized` instead of `ISerializable`, but then all classes having this type to serialize would have to implement `IDeserializationCallback.OnDeserialization` for ensuring this type `OnDeserialized` has already been executed. So I have kept `ISerializable`.